code/game/machinery/camera/camera_assembly.dm

```diff
@@ -105,6 +105,8 @@
 	if(is_type_in_list(W, possible_upgrades) && !is_type_in_list(W, upgrades)) // Is a possible upgrade and isn't in the camera already.
 		if(!user.unEquip(W))
 			return
+		if(!W.drop_item()) // no eating borg items
+			return

```

Comment:
Just replace the unEquip line with:
```
if(user.drop_item())
	return
```

And replace `W.loc = src` with `W.forceMove(src)`.


In camera/attackby() add `if(user.drop_item())` before calling upgradeXRay() in https://github.com/tgstation/tgstation/blob/master/code/game/machinery/camera/camera.dm#L166